apps/api/src/services/search.ts

```diff
@@ -126,12 +130,14 @@ export class SearchService {
     const instructors = await this.instructorMappingFromResults(results);
     return {
       count: results.size,
-      results: Array.from(results.entries())
+      results: results
+        .entries()
         .map(([key, rank]) => ({
           type: "instructor" as const,
           result: getFromMapOrThrow(instructors, key),
           rank,
         }))
+        .toArray()
         .slice(input.skip, input.skip + input.take),
```

Comment:
Instead of doing this, maybe consider using [`Iterator.drop`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/drop) and [`Iterator.take`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/take), since that way we can avoid having the iterator return elements we don't want.

Reply:
Epic, did not know that was a thing :star_struck: